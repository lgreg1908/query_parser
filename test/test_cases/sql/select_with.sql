WITH cte AS (
    SELECT col1 FROM table1
    UNION
    SELECT col2 FROM table2
    )
    SELECT * FROM cte