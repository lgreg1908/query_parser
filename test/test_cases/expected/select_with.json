{
    "type": "ROOT",
    "children": [
        {"type": "CTE", "value": "WITH", "is_group": true, "children": [
            {"type": "Identifier", "value": "cte", "is_group": false},
            {"type": "Keyword", "value": "AS", "is_group": false},
            {"type": "Subselect", "value": "(...)", "is_group": true, "children": [
                {"type": "DML", "value": "SELECT", "is_group": false},
                {"type": "Identifier", "value": "col1", "is_group": false},
                {"type": "Keyword", "value": "FROM", "is_group": false},
                {"type": "Identifier", "value": "table1", "is_group": false},
                {"type": "SetOperation", "value": "UNION", "is_group": false},
                {"type": "DML", "value": "SELECT", "is_group": false},
                {"type": "Identifier", "value": "col2", "is_group": false},
                {"type": "Keyword", "value": "FROM", "is_group": false},
                {"type": "Identifier", "value": "table2", "is_group": false}
            ]}
        ]},
        {"type": "DML", "value": "SELECT", "is_group": false},
        {"type": "Wildcard", "value": "*", "is_group": false},
        {"type": "Keyword", "value": "FROM", "is_group": false},
        {"type": "Identifier", "value": "cte", "is_group": false}
    ]
}